omnigibson/sensors/vision_sensor.py

```diff
@@ -287,9 +288,14 @@ def _get_obs(self):
             reordered_modalities = self._modalities
 
         for modality in reordered_modalities:
-            raw_obs = self._annotators[modality].get_data()
+            raw_obs = self._annotators[modality].get_data(device="cuda")
             # Obs is either a dictionary of {"data":, ..., "info": ...} or a direct array
             obs[modality] = raw_obs["data"] if isinstance(raw_obs, dict) else raw_obs
+            # All segmentation modalities return uint32 warp arrays, but PyTorch doesn't support it
+            if modality in ["seg_semantic", "seg_instance", "seg_instance_id"]:
+                obs[modality] = obs[modality].view(lazy.warp.int32)
```

Comment:
As we discussed before, the pipeline here can be:

On GPU: get the warp tensor from annotator by get_dataing with GPU, convert to torch using the warp util - in that case we need the synchronize call
On CPU: get the CPU numpy tensor from annotator by get_dataing with CPU, convert zero-cost to torch using torch.from_numpy (note that this shares underlying buffer)

In that case do we even need the customized annotator class?

If you think this is tricky let's open a P0 ticket about this (high speed impact) and merge as-is and follow up immediately after.

Reply:
Please check this commit: https://github.com/StanfordVL/OmniGibson/pull/780/commits/8281cca9dae7730f71140c169e81ca8607078312

*This wasn't leaking the warp array. It would have been converted to a torch tensor in the next two lines.